src/base/dof_map.C

```diff
@@ -355,7 +355,7 @@ void DofMap::set_nonlocal_dof_objects(iterator_type objects_begin,
         requested_ids[obj->processor_id()].push_back(obj->id());
     }
 #ifdef DEBUG
-  for (processor_id_type p=0; p != this->n_processors(); ++p)
+  for (processor_id_type p=0, np = this->n_processors(); p != np; ++p)
```

Comment:
I like this idiom. I believe the only restriction is that both variables have to be the same type?

Reply:
👍 for `for (auto n : elem->node_index_range())` code, I ran into a few places when refactoring for loops where needing an index made things kind of annoying.